src/main/kotlin/hs/kr/equus/feed/domain/question/service/UpdateQuestionService.kt

```diff
@@ -0,0 +1,33 @@
+package hs.kr.equus.feed.domain.question.service
+
+import hs.kr.equus.feed.domain.question.domain.repository.QuestionRepository
+import hs.kr.equus.feed.domain.question.exception.FeedWriterMismatchException
+import hs.kr.equus.feed.domain.question.exception.QuestionNotFoundException
+import hs.kr.equus.feed.domain.question.presentation.dto.request.UpdateQuestionRequest
+import hs.kr.equus.feed.global.utils.user.UserUtils
+import org.springframework.data.repository.findByIdOrNull
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import java.util.UUID
+
+@Service
+class UpdateQuestionService(
+    private val questionRepository: QuestionRepository,
+    private val userUtils: UserUtils
+) {
+    @Transactional
+    fun execute(questionId: UUID, updateQuestionRequest: UpdateQuestionRequest) {
+        val user = userUtils.getCurrentUser()
+        val question = questionRepository.findByIdOrNull(questionId) ?: throw QuestionNotFoundException
+
+        if (question.userId != user.id) {
+            throw FeedWriterMismatchException
+        }
+
+        question.updateQuestion(
+            title = updateQuestionRequest.title,
+            content = updateQuestionRequest.content,
+            isPublic = updateQuestionRequest.isPublic
+        )
```

Comment:
with이나 run같은 scope function 쓰면 어떨까요?
```suggestion
        updateQuestionRequest.run {
            question.updateQuestion(
                title = title,
                content = content,
                isPublic = isPublic
            )
        }
```

Reply:
위 내용과 같이, 코드를 이런식으로 짜는건 어떤가요?
```
fun validateWriter(user: User) {
        if (this.userId != user.id) {
            throw FeedWriterMismatchException
        }
}

fun updateQuestion(user: User, updateQuestionRequest: UpdateQuestionRequest) {
        validateOwner(user)
        
        this.title = updateQuestionRequest.title
        this.content = updateQuestionRequest.content
        this.isPublic = updateQuestionRequest.isPublic
}
```

Reply:
validate를 update 함수 안에 넣는건 좋아요! 근데 Domain 객체가 dto에 의존하게 만드는건 구조적으로 좀 안 좋은 것 같습니다. request dto는 request 용도로만 사용하는게 역할 분리 측면에서 깔끔한듯 합니다